lib/spack/spack/fetch_strategy.py

```diff
@@ -478,6 +478,7 @@ def __init__(self, **kwargs):
 
         # Set a URL based on the type of fetch strategy.
         self.url = kwargs.get(self.url_attr, None)
+        self.clean = kwargs.get("clean", True)
```

Comment:
I would give a more specific name to this argument, like `archive_vcs_info`. What do you think?